src/traits.rs

```diff
@@ -0,0 +1,23 @@
+// https://doc.rust-lang.org/src/core/convert.rs.html#478-486
+pub trait Translate<T>: Sized {
+    type Error;
+
+    fn translate(value: T) -> Result<Self, Self::Error>;
```

Comment:
TODO: This definition should take an optional Config structure that we can pass around as context.